src/wwwroot/app.js

```diff
@@ -266,10 +266,9 @@
     var pageOpenForAll = openForAllUrls.includes(currentPath);
     var pageRequireLogout = requireLogoutUrls.includes(currentPath);
     var pageRequireTemporalAuth = requireTemporalAuthUrls.includes(currentPath);
-    var tokenPermissons = auth.decodeToken();
         
-    if(tokenPermissons && !currentPath.match(tokenPermissons.acceptedUrlsPattern)){
-      $location.path($route.routes[tokenPermissons.defaultUrl].originalPath);
+    if(!auth.isUrlAllowed(currentPath)) {
+      $location.path($route.routes[auth.getDefaultUrl()].originalPath);
```

Comment:
I know that it does not happen, but since it is legal that _getDefaultUrl()_ returns _null_ I think that we should take into account that case here.

Reply:
For example:

```js
if (!auth.isUrlAllowed(currentPath)) {
    $location.path(auth.getDefaultUrl() || '/reports');
```

Also, I am not sure why do you search the route in _$route.routes_. I mean, if the route exist it should be true_ `x
$route.routes[x].originalPath`, Right? Am I missing something?

Reply:
Regarding `x == $route.routes[x].originalPath`, it is right, but when there is no default route (when `x == null`) we want to redirect to _global default URL_, @mvillaseco is changing the code to make that behavior more clear.